### PR TITLE
moving javascripts to the page footer

### LINKED ISSUE #1455: move scripts to the bottom of the page
https://github.com/OpenGrok/OpenGrok/blob/master/web/httpheader.jspf#L78

scripts should be sourced in as last on page
(performance best practice)

### LINKED ISSUE #1490: Navigate window expects scopes window to be ready
This function in utils.js expects scopes window to be inizialized. The check should contain also that condition when the scopes window isnt available at all.

```js
updatePosition: function ($w) {
    var a = {}
    a.top = $.scopesWindow.is(':visible') ? $.scopesWindow.offset().top + $.scopesWindow.outerHeight() + 20 : this.getTopOffset() + 10;
    a.height = Math.min(parseFloat($w.css('max-height')) || 480, $(browserWindow).outerHeight() - a.top - ($w.outerHeight(true) - $w.height()) - 20);

    if (a.height == $w.height() && a.top == this.getTopOffset())
        return $w;

    if (this.$content.children().length === 0)
        // the window is empty
        delete a.height

    return $w.stop().animate(a)
},
```

### PR DESCRIPTION
**Pull request by tulinkry**
fixes #1455
fixes #1490
related to #1488

I double checked all jsps; this configuration appears to work.

Summary:
1. for the `document` variables I used functions so that the inner block does not need to be defined in advance
2. moved the js to footer
3. fixed observed problem described in #1490 
4. merged `$(document).ready()` calls to single call

### CODE REVIEW

**Comment by tarzanek** [comment]
;) sleeps in tests made travis cry?

**Comment by tarzanek** [comment]
I have restarted the build

**Comment by tulinkry** [comment]
Aargh. Need to update them after the release.

**Comment by tarzanek** [comment]
no prob, we've all been there and done that
sleeps are just evil ...

File: web/repos.jspf [hunk 1]
```diff
@@ -44,7 +44,6 @@ Copyright (c) 2015, 2017, Oracle and/or its affiliates. All rights reserved.
 <%@page import="org.opensolaris.opengrok.configuration.Group"%>
 <%@page import="java.util.List"%>
 <%@page import="org.opensolaris.opengrok.web.PageConfig"%>
-<script src="<%=request.getContextPath()%>/js/repos-0.0.1.js" type="text/javascript"></script>
```
Comment (vladak) [inline web/repos.jspf]: pls add a comment to whoever is including `repos.jspf` to add the `script` tag at the end of the file

Comment (tulinkry) [inline web/repos.jspf]: added a comment to the top

**Comment by tarzanek** [comment]
thank you @tulinkry ! merging

**Pull request merged**